sql: trace schema changes

tbg opened the pull request:
Motivated by the fact that previously, `DROP TABLE` did not allow any
kind of introspection. Now, at least there's something you can look
at in `/debug/requests` to see activity.

Touches #19004

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/20491)
<!-- Reviewable:end -->

tschottdorf commented on pkg/sql/schema_changer.go:
```diff
@@ -1103,6 +1111,10 @@ func (s *SchemaChangeManager) Start(stopper *stop.Stopper) {
 						sc.execAfter = timeutil.Now().Add(delay)
 					}
+
+					// TODO(vivek): shouldn't we only `break` if we actually ran a schema change
```
@vivekmenezes could you comment on this?

vivekmenezes replied:
I don't think this is important.

tschottdorf replied:
Can you be more specific? I.e. explain why we even pace schema changes in that way (why not do away with `execAfter`)? I'm happy to add commentary once you do.

vivekmenezes replied:
I believe the only reason to pace these is to allow different schema changes to execute on different nodes and distribute the load. But that was a thought a long time ago. You can safely move the break inside the if statement.

vivekmenezes commented on pkg/sql/schema_changer.go:
```diff
@@ -927,7 +929,8 @@ type SchemaChangeManager struct {
 func NewSchemaChangeManager(
-	st *cluster.Settings,
+	ambientCtx log.AmbientContext,
+	_ *cluster.Settings,
```
What is this about?

vivekmenezes reviewed (approved):
Thanks for making this change!

tbg committed:
sql: trace schema changes

Motivated by the fact that previously, `DROP TABLE` did not allow any
kind of introspection. Now, at least there's something you can look
at in `/debug/requests` to see activity.

Touches #19004

Release note: None

tschottdorf commented:
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[pkg/sql/schema_changer.go, line 933 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/20491#-L-bzL9k3l4dx4zdQutv-r1-933:-L-gFQuAJcaTD9_m_jAl:bxc0c4r) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4d41dedf3e0ba7a4b46a2f258e8bc8fdfccc1a3/pkg/sql/schema_changer.go#L933)):*
<details><summary><i>Previously, vivekmenezes wrote…</i></summary><blockquote>

What is this about?
</blockquote></details>

The argument was unused. I originally left it in because it's nice to have access to `Settings` everywhere but removed it now.

---

*[pkg/sql/schema_changer.go, line 1115 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/20491#-L-bzL9k3l4dx4zdQutv-r1-1115:-L-gG2U2P8_mndr_SX7T:b-kcqmh9) ([raw file](https://github.com/cockroachdb/cockroach/blob/f4d41dedf3e0ba7a4b46a2f258e8bc8fdfccc1a3/pkg/sql/schema_changer.go#L1115)):*
<details><summary><i>Previously, vivekmenezes wrote…</i></summary><blockquote>

I believe the only reason to pace these is to allow different schema changes to execute on different nodes and distribute the load. But that was a thought a long time ago. You can safely move the break inside the if statement.
</blockquote></details>

Thanks, I moved the `break` in and added a `TODO` to evaluate removal of the pacing.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20491)*
<!-- Sent from Reviewable.io -->
